toolkits/global/packages/global-article/js/reading-companion.js

```diff
@@ -1,522 +1,527 @@
-					return;
-				}
+		function buildTabBar(tabs) {
+			return '<ul class="' + READING_COMPANION_CLASS + '__tabs" role="tablist">' + tabs.join('') + '</ul>';
+		}
 
-				var tabCount = buildTabs();
-				if (tabCount === 0) {
-					return;
-				}
+		// eslint-disable-next-line unicorn/consistent-function-scoping
+		function insert(element, html) {
+			element.insertAdjacentHTML('afterbegin', html);
+		}
+
+		// eslint-disable-next-line unicorn/consistent-function-scoping
+		function insertBefore(element, html) {
+			element.insertAdjacentHTML('beforebegin', html);
+		}
 
-				bindEvents(scheduler, emitter);
-				if (tabCount > 1) {
-					switchToTab('sections');
+		function buildTabs() {
+			var tabs = ['sections', 'figures', 'references'].map(function (name) {
+				var container = document_.querySelector('.' + READING_COMPANION_CLASS + '__' + name);
+				var html = htmlFor(name, select(document_, '.js-' + READING_COMPANION_CLASS + '-' + name + '-item'));
+				// eslint-disable-next-line unicorn/prefer-string-slice
+				var tab = htmlForTab(name, name.charAt(0).toUpperCase() + name.substring(1));
+
+				if (html && container) {
+					container.setAttribute('aria-labelledby', 'tab-' + name);
+					insert(container, '<div class="' + READING_COMPANION_CLASS + '__scroll-pane">' + html + '</div>');
+				} else if (container) {
+					// eslint-disable-next-line unicorn/prefer-node-remove
+					container.parentNode.removeChild(container);
 				}
 
-				initialised = true; // eslint-disable-line no-use-before-define
-			};
+				return (html && container) ? tab : false;
+			}).filter(function (tab) {
+				return Boolean(tab);
+			});
 
-			var doSetup = function (mql) {
-				setup(mql);
-				emitter.emit('rc.display', !mql.matches && initialised);
-			};
+			var firstTabPanel = _container.querySelector('.' + READING_COMPANION_CLASS + '__panel');
 
-			var mq = win.matchMedia(config.matchMediaQuery || '(max-width: 639px)');
-			mq.addListener(doSetup);
-			doSetup(mq);
+			var tabCount = tabs.length;
+			if (tabCount > 1) {
+				insertBefore(firstTabPanel, buildTabBar(tabs));
+			} else if (tabCount === 1) {
+				insertBefore(firstTabPanel, '<h3 class="' + READING_COMPANION_CLASS + '__heading">Sections</h3>');
+			}
+			return tabCount;
 		}
+
+		return {
+			init: function (config, scheduler, emitter) {
+				_access = Boolean(config.access);
+				_offset = config.offset || 0;
+
+				var initialised = false;
+
+				var setup = function (mql) {
+					if (initialised || mql.matches) { // eslint-disable-line no-use-before-define
+						return;
+					}
+
+					_container = document_.querySelector('[' + COMPONENT_ATTR + '=reading-companion-sticky]');
+
+					if (!_container) {
+						return;
+					}
+
+					var tabCount = buildTabs();
+					if (tabCount === 0) {
+						return;
+					}
+
+					bindEvents(scheduler, emitter);
+					if (tabCount > 1) {
+						switchToTab('sections');
+					}
+
+					initialised = true; // eslint-disable-line no-use-before-define
```

Comment:
Same thing here - can we remove the eslint comment?

Reply:
done!